app/src/main/java/io/gnosis/safe/ui/safe/add/SafeInputView.kt

```diff
@@ -0,0 +1,66 @@
+package io.gnosis.safe.ui.safe.add
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.LinearLayout
+import androidx.core.content.res.ResourcesCompat
+import io.gnosis.safe.R
+import io.gnosis.safe.databinding.ViewSafeInputBinding
+import io.gnosis.safe.utils.formatEthAddress
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.visible
+
+class SafeInputView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    private val binding = ViewSafeInputBinding.inflate(LayoutInflater.from(context), this)
+
+    var address: Solidity.Address? = null
```

Comment:
there is no need to use Solidity.Address here. address field could be just a String. There is anyway no validation here and error state is set to false regardless if address is a correct safe address. I think it is a good idea for this view to be dumb and just display stuff. All business logic & validation should be in view model

Reply:
Well, I need it for the blockies as an Address. If the string is an invalid address, that process should happen outside this view. In the current state I would say this view is as dumb as it can be. It either displays an address or an error label.

Reply:
having it as a string will alllow to potentially set values in xml. You can convert into solidity in the view itself.